### PR TITLE
CORTX-33416: Append Machine ID in CSM log path

### PR DESCRIPTION
**Pull request by abhijit1patil**
Signed-off-by: abhijit1patil <abhijit.1.patil@seagate.com>

# Pull Request
## Problem Statement
-   [CORTX-33416](https://jts.seagate.com/browse/CORTX-33416)

## Design
-   Added machine id in log path while initiating log in csm agent

-   \[ \] Coding conventions are followed and code is consistent

-   \[ \] Unit and System Tests are added
-   \[ \] Test Cases cover Happy Path, Non-Happy Path and Scalability
-   \[ \] Testing was performed with RPM

-   \[ \] Interface change (if any) are documented
-   \[ \] Side effects on other features (deployment/upgrade)
-   \[ \] Dependencies on other component(s)

-   \[ \] JIRA number/GitHub Issue added to PR
-   \[ \] PR is self reviewed
-   \[ \] Jira and state/status is updated and JIRA is updated with PR link
-   \[ \] Check if the description is clear and explained

-   \[ \] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by abhijit1patil** [comment]
## Log hierarchy:- 
### /mnt/fs-local-volume/local-path-provisioner/pvc-72112d16-c072-4922-bc38-19907ab361e4_cortx_cortx-control-fs-local-pvc-cortx/log/

├── csm
│   ├── csm_setup.log
│   └── e7c1a627bd659812f467399461b2ce7e
│       └── csm_agent.log
├── provisioner
│   └── e7c1a627bd659812f467399461b2ce7e
│       └── cortx_setup.log
└── utils
    └── e7c1a627bd659812f467399461b2ce7e
        ├── support
        │   ├── support_bundle.log
        │   └── support_bundle_node.log
        └── utils_setup.log

7 directories, 6 files

File: csm/core/agent/csm_agent.py [hunk 1]
```diff
@@ -44,10 +44,13 @@ def init():
         file_size_in_mb = Conf.get(const.CSM_GLOBAL_INDEX, "Log>file_size")
         log_level = "DEBUG" if Options.debug else Conf.get(const.CSM_GLOBAL_INDEX, "Log>log_level")
         console_output = Conf.get(const.CSM_GLOBAL_INDEX, "Log>console_logging") == "true"
+        csm_log_path = Conf.get(const.CSM_GLOBAL_INDEX, "Log>log_path")
+        machine_id = Conf.machine_id
+        csm_machine_log_path = f"{csm_log_path}/{machine_id}"
```
Comment (pawankumarsrivastava) [inline csm/core/agent/csm_agent.py:49]: Please add it for CSM setup logs too

File: csm/core/agent/csm_agent.py [hunk 2]
```diff
@@ -44,10 +44,13 @@ def init():
         file_size_in_mb = Conf.get(const.CSM_GLOBAL_INDEX, "Log>file_size")
         log_level = "DEBUG" if Options.debug else Conf.get(const.CSM_GLOBAL_INDEX, "Log>log_level")
         console_output = Conf.get(const.CSM_GLOBAL_INDEX, "Log>console_logging") == "true"
+        csm_log_path = Conf.get(const.CSM_GLOBAL_INDEX, "Log>log_path")
+        machine_id = Conf.machine_id
```
Comment (pawankumarsrivastava) [inline csm/core/agent/csm_agent.py:48]: Check Null for machine ID and add Warn log and do not append the path

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: csm/core/agent/csm_agent.py (hunk 1 above)
Comment (abhijit1patil) [reply csm/core/agent/csm_agent.py:49]: updated

File: csm/core/agent/csm_agent.py (hunk 2 above)
Comment (pawankumarsrivastava) [reply csm/core/agent/csm_agent.py:48]: ???

File: csm/core/agent/csm_agent.py (hunk 1 above)
Comment (pawankumarsrivastava) [reply csm/core/agent/csm_agent.py:49]: For CSM setup I do not see the changes

Comment (abhijit1patil) [reply csm/core/agent/csm_agent.py]: Please check now

File: csm/core/agent/csm_agent.py (hunk 2 above)
Comment (abhijit1patil) [reply csm/core/agent/csm_agent.py]: Pushing the change.. 
I can not add warning logs as we are initializing logs after this

**Review by pawankumarsrivastava — APPROVED** [review]

**Comment by walterlopatka** [comment]
Hi @abhijit1patil,

While this code change is fairly straightforward and looks good as far as it goes, but I think this is an unnecessary change.

As part of the feature we are working on (high availability cortx-control pods), I am changing the volume that backs the cortx-control pod to a K8s "emptyDir".  This is ephemeral storage that lives with the pod, is accessed only by containers in the pod, and is torn down at the end of the pod's lifecycle.  So there is no way that different pods (even on the same node) will ever access the same /etc/cortx directory (for logging, config, or anything else.

In fact, even without this emptyDir change, each cortx-control pod would allocate a different PVC so that, again, there would not be a conflict even in that case.  On the local node, each of the PVCs will be mapped to individual directories for each separate pod/volume.

To illustrate, this is what the local PV directories look like on one of my K8s worker nodes:
```bash
[root@ssc-vm-g2-rhev4-3362 ~]# cd /mnt/fs-local-volume/local-path-provisioner/
[root@ssc-vm-g2-rhev4-3362 local-path-provisioner]# ll
total 24
drwxrwxrwx 9 root root 4096 Jul 21 11:52 pvc-1d2d0fb6-c294-4094-8643-6b8a6db426f7_default_data-cortx-server-1
drwxrwxrwx 8 root root 4096 Jul 21 11:52 pvc-1e764ed1-1c22-4115-b505-915a5b5dead4_default_data-cortx-data-g1-2
drwxrwxrwx 8 root root 4096 Jul 21 11:53 pvc-517cd94f-3350-41f8-9efa-e6b9202dcde4_default_data-cortx-data-g0-4
drwxrwxrwx 3 root root 4096 Jul 21 11:51 pvc-ca3f52a7-2171-4c56-8815-a9fbe604bfa6_default_data-cortx-zookeeper-1
drwxrwxrwx 4 root root 4096 Jul 21 11:52 pvc-d150475a-4b24-4bfb-84eb-2e2186a6a8bd_default_data-default-cortx-consul-server-0
drwxrwxrwx 4 root root 4096 Jul 21 11:51 pvc-ed8ca4ab-86cd-498c-95c1-a7db47d7c2da_default_data-cortx-kafka-1
```
If there were multiple cortx-control pods on this node then the data for each would be in a separate directory.

(I am adding the same comment to Jira for reference.)

**Review by Pranali04796 — CHANGES_REQUESTED** [review]

**Comment by abhijit1patil** [comment]
> 

@kupranay @pawankumarsrivastava Could you reply to @walterlopatka's comment?

File: csm/conf/setup.py [hunk 3]
```diff
@@ -234,7 +234,12 @@ def setup_logs_init():
     @staticmethod
     def get_csm_log_path():
         log_path = Conf.get(const.CONSUMER_INDEX, const.CORTX_LOG_PATH_KEY)
-        return f"{log_path}/{const.CSM_COMPONENT_NAME}"
+        if Conf.machine_id is not None:
+            machine_id = Conf.machine_id
+            csm_machine_log_path = f"{log_path}/{const.CSM_COMPONENT_NAME}/{machine_id}"
+        else:
+            csm_machine_log_path = f"{log_path}/{const.CSM_COMPONENT_NAME}"
+        return csm_machine_log_path
```
[the remark this answers is not in the document]
Comment (abhijit1patil) [reply csm/conf/setup.py]: I have updated my code already.. please check
I think you commented on old commit

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]
handle pranali's comment

**Comment by abhijit1patil** [comment]
As per suggestion from Walter,
Changes from this PR are not required

**Pull request closed without merging**